File: LeetCode/9.Palindrome_Number/Mario/9.Palindrome_Number.java
class Solution {
    public boolean isPalindrome(int x) {
        int origin = x;
        int rev = 0;
        while (x > 0) {
            rev = rev * 10 + x % 10;
            x /= 10;
        }
        return rev == origin;
    }
}

File: LeetCode/9.Palindrome_Number/9.Palindrome_Number.swift
class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        if x == 0 { return true }
        if x < 0 || x % 10 == 0 {
            return false
        }
        var left = x, right = 0
        while left > right {
            right = right * 10 + left % 10
            left /= 10
        }

        return left == right || left == right / 10
    }
}
